client/src/graphs/svg/util.js

```diff
@@ -23,7 +24,7 @@ export const traverse = (root, callBackFn, depth = 0) => {
   callBackFn(root, depth);
   if (root.nodes) {
     const d = depth + 1;
-    for (let i = 0; i < root.nodes.length; i++) {
+    for (let i = 0; i < root.nodes?.length; i++) {
```

Comment:
You might get an issue if `root.nodes?.length` returns `undefined`.
Why not using [forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)?
```JS
root.nodes?.forEach(node => traverse(node, callBackFn, depth + 1));
```

Reply:
Done.